Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSessionCocoa.mm

```diff
@@ -47,20 +50,74 @@
+        switch (disposition) {
+        case AuthenticationChallengeDisposition::UseCredential: {
+            if (!credential.isEmpty())
+                completion(true);
+        }
+        FALLTHROUGH;
+        case AuthenticationChallengeDisposition::RejectProtectionSpaceAndContinue:
+        case AuthenticationChallengeDisposition::PerformDefaultHandling: {
+            OSStatus status = SecTrustEvaluateAsyncWithError(secTrust.get(), dispatch_get_main_queue(), makeBlockPtr([completion = completion](SecTrustRef trustRef, bool result, CFErrorRef error) {
+                completion(result);
+            }).get());
+            if (status != errSecSuccess)
+                completion(false);
+            return;
+        }
+        case AuthenticationChallengeDisposition::Cancel:
+            completion(false);
+            return;
+        }
+        RELEASE_ASSERT_NOT_REACHED();
+    };
+
+    auto* sessionCocoa = static_cast<NetworkSessionCocoa*>(connectionToWebProcess->networkProcess().networkSession(connectionToWebProcess->sessionID()));
+
+    if (sessionCocoa && sessionCocoa->fastServerTrustEvaluationEnabled()) {
+        auto decisionHandler = makeBlockPtr([sessionCocoa = WeakPtr { *sessionCocoa }, connectionToWebProcess = WTFMove(connectionToWebProcess), pageID = WTFMove(pageID), clientOrigin = WTFMove(clientOrigin), challengeCompletionHandler = WTFMove(challengeCompletionHandler)](NSURLAuthenticationChallenge *challenge, OSStatus trustResult) mutable {
+            auto* session = sessionCocoa.get();
+            if (trustResult == noErr || !session) {
+                challengeCompletionHandler(AuthenticationChallengeDisposition::PerformDefaultHandling, WebCore::Credential());
+                return;
+            }
+
+            connectionToWebProcess->networkProcess().protectedAuthenticationManager()->didReceiveAuthenticationChallenge(connectionToWebProcess->sessionID(), pageID, &clientOrigin.topOrigin, challenge, NegotiatedLegacyTLS::No, WTFMove(challengeCompletionHandler));
+        });
+
+
+        [NSURLSession _strictTrustEvaluate:challenge.get() queue:[NSOperationQueue mainQueue].underlyingQueue completionHandler:decisionHandler.get()];
+        return;
+    }
+
+    connectionToWebProcess->networkProcess().protectedAuthenticationManager()->didReceiveAuthenticationChallenge(connectionToWebProcess->sessionID(), pageID, &clientOrigin.topOrigin, challenge.get(), NegotiatedLegacyTLS::No, WTFMove(challengeCompletionHandler));
+}
+
+static RetainPtr<nw_parameters_t> createParameters(NetworkConnectionToWebProcess& connectionToWebProcess, URL&& url, WebKit::WebPageProxyIdentifier&& pageID, struct WebCore::ClientOrigin&& clientOrigin)
 {
-    auto configureWebTransport = [origin = WTFMove(origin)](nw_protocol_options_t options) {
+    auto configureWebTransport = [clientOrigin = clientOrigin](nw_protocol_options_t options) {
```

Comment:
clientOrigin = clientOrigin.clientOrigin.toString()  Then we copy less stuff.  And adjust the line below to keep it compiling.